internal/lookout/repository/query.go

```diff
@@ -50,3 +58,285 @@ func (r *SQLJobRepository) GetQueueStats() ([]*lookout.QueueInfo, error) {
 	}
 	return result, nil
 }
+
+func (r *SQLJobRepository) GetJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*lookout.JobInfo, error) {
+	rows, err := r.queryJobsInQueue(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	result := parseJobsInQueueRows(rows)
+	return result, nil
+}
+
+type jobsInQueueRow struct {
+	JobId     string
+	Queue     string
+	Owner     string
+	JobSet    string
+	Priority  sql.NullFloat64
+	Submitted pq.NullTime
+	Cancelled pq.NullTime
+	JobJson   sql.NullString
+	RunId     sql.NullString
+	Cluster   sql.NullString
+	Node      sql.NullString
+	Created   pq.NullTime
+	Started   pq.NullTime
+	Finished  pq.NullTime
+	Succeeded sql.NullBool
+	Error     sql.NullString
+}
+
+func (r *SQLJobRepository) queryJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*jobsInQueueRow, error) {
+	query, args, err := makeGetJobsInQueueQuery(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := r.db.Query(query, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	joinedRows := make([]*jobsInQueueRow, 0)
+	for rows.Next() {
+		row := &jobsInQueueRow{}
+		err := rows.Scan(
+			&row.JobId,
+			&row.Owner,
+			&row.JobSet,
+			&row.Priority,
+			&row.Submitted,
+			&row.Cancelled,
+			&row.JobJson,
+			&row.RunId,
+			&row.Cluster,
+			&row.Node,
+			&row.Created,
+			&row.Started,
+			&row.Finished,
+			&row.Succeeded,
+			&row.Error,
+		)
+		if err != nil {
+			return nil, err
+		}
+		joinedRows = append(joinedRows, row)
+	}
+	return joinedRows, nil
+}
+
+func makeGetJobsInQueueQuery(opts *lookout.GetJobsInQueueRequest) (string, []interface{}, error) {
+	dialect := goqu.Dialect("postgres")
+
+	subDs := dialect.
+		From("job").
+		LeftJoin(goqu.T("job_run"), goqu.On(goqu.Ex{
+			"job.job_id": goqu.I("job_run.job_id"),
+		})).
+		Select(goqu.I("job.job_id"))
+
+	subDs = addWhere(subDs, opts.Queue, opts.JobSetIds)
+
+	subDs = subDs.GroupBy(goqu.I("job.job_id"))
+
+	subDs = addHaving(subDs, opts.JobStates)
```

Comment:
These `addHaving` and `addWhere` are quite hard to follow. Could you change it to have method like `CreateJobStateFilter(opts.JobSetIds)` and call it here like `ds.Having(CreateJobStateFilter(opts.JobStates))`. That way it will be much easier to see how the query actually looks like just from this method.